tests/integrations/base.py

```diff
@@ -104,3 +110,68 @@ def assert_new_commit(self, steps=1):
     def assert_file_content(self, file_path, content):
         with open(self.repo_path + "/" + file_path) as f:
             assert f.read() == content
+
+
+class GitFSLog(object):
+    def __init__(self, file_descriptor):
+        self._partial_line = None
+        self.line_buffer = collections.deque()
+        self.file_descriptor = file_descriptor
+
+    def get_line(self):
+        if not self.line_buffer:
+            self._read_data()
+        if self.line_buffer:
+            return self.line_buffer.popleft()
+
+    def _read_data(self):
+        data = os.read(self.file_descriptor, 2048).splitlines(True)
+        if not data:
+            return
+        if self._partial_line:
+            data[0] = self._partial_line + data[0]
+        if not data[-1].endswith("\n"):
+            self._partial_line = data[-1]
+            data = data[:-1]  # discard the partial line
+        else:
+            self._partial_line = None
+        self.line_buffer.extend(data)
+
+    def clear(self):
+        while os.read(self.file_descriptor, 2048):
```

Comment:
The intent was to consume all available input, but i guess i should do that with `seek`
So, when you want to expect a certain output, you want to make sure you're not matching past output, so you'll advance to the end of the stream, do something with a side effect, then expect the line.